omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptReferenceTermSearchHandler1_9.java

```diff
@@ -62,41 +68,98 @@ public PageableResult search(RequestContext context) throws ResponseException {
 		String source = context.getParameter("source");
 		String code = context.getParameter("code");
 		String name = context.getParameter("name");
-		
+		String searchType = context.getParameter("searchType");
+
+
 		ConceptSource conceptSource = conceptService.getConceptSourceByUuid(source);
-		
+
 		if (conceptSource == null) {
 			conceptSource = conceptService.getConceptSourceByName(source);
 		}
-		
+
 		if (conceptSource == null) {
 			return new EmptySearchResult();
 		}
-		
+
 		if (code != null || name != null) {
-			ConceptReferenceTerm term = null;
-			if (code != null) {
-				term = conceptService.getConceptReferenceTermByCode(code, conceptSource);
-				if (name != null && !name.equals(term.getName())) {
-					term = null;
-				}
-			} else {
-				term = conceptService.getConceptReferenceTermByName(name, conceptSource);
-			}
-			
-			if (term == null || (term.isRetired() && !context.getIncludeAll())) {
-				return new EmptySearchResult();
+			if(SEARCH_TYPE_STARTS_WITH.equals(searchType)){
+				return searchStartsWith(context, code, name, conceptSource);
 			} else {
-				return new AlreadyPaged<ConceptReferenceTerm>(context, Arrays.asList(term), false);
+				return searchEquals(context, code, name, conceptSource);
 			}

```

Comment:
Please throw  ResponseException if searchType is not valid (neither equals nor startsWith)